solver/cacheopts.go

```diff
@@ -10,6 +10,8 @@ import (
 
 type CacheOpts map[interface{}]interface{}
 
+type ProgressKey struct{}
```

Comment:
It's a bit weird construction to have this as a public type here. I think it would be better to define it as a public function that takes in context and returns progress controller, leaving the key type private to solver pkg.

Could also do it in `progress/controller` package and define `FromContext`/`WithController` functions there.

Reply:
Agreed, updated with it in `solver/cacheopts.go`. Putting it in `progress/controller` would have been nice but would have made that package depend on `solver` (in order to use the cache opt getter).